app/controllers/Authentication.scala

```diff
@@ -118,25 +124,49 @@ class Authentication @Inject() (
     Ok(views.html.main()(Html("")))
   }
 
+  def normalizeName(name: String):Option[String] = {
+    val replacementMap = Map("ü" -> "ue", "Ü" -> "Ue", "ö" -> "oe", "Ö" -> "Oe", "ä" -> "ae", "Ä" -> "Ae", "ß" -> "ss",
+      "é" -> "e", "è" -> "e", "ê" -> "e", "È" -> "E", "É" -> "E", "Ê" -> "E",
+      "Ç" -> "C", "ç" -> "c", "ñ" -> "n", "Ñ" -> "N", "ë" -> "e", "Ë" -> "E", "ï" -> "i", "Ï" -> "I",
+      "å" -> "a", "Å" -> "A", "œ" -> "oe", "Œ" -> "Oe", "æ" -> "ae", "Æ" -> "Ae",
+      "þ" -> "th", "Þ" -> "Th", "ø" -> "oe", "Ø" -> "Oe", "í" -> "i", "ì" -> "i")
+
+    val finalName = name.map(c => replacementMap.getOrElse(c.toString,c.toString)).mkString.replaceAll("[^A-Za-z0-9_\\-\\s]", "")
+    if(finalName.isEmpty)
+      None
+    else
+      Some(finalName)
+  }
+
   def handleRegistration = Action.async { implicit request =>
-    signUpForm.bindFromRequest.fold(
-      bogusForm =>  Future.successful(BadRequest(bogusForm.toString)),
+      signUpForm.bindFromRequest.fold(
+      bogusForm => Future.successful(BadRequest(bogusForm.toString)),
       signUpData => {
         val loginInfo = LoginInfo(CredentialsProvider.ID, signUpData.email)
+        var errorMap = List[String]()
```

Comment:
is the name intentional? it’s not a map, right? how about »errors«?

Reply:
At first it was a map but you're right the name is misleading. Will change it